packages/iov-bns/src/encode.ts

```diff
@@ -158,6 +177,23 @@ function buildSwapTimeoutTx(tx: SwapTimeoutTx): codecImpl.app.ITx {
   };
 }
 
+function buildRegisterBlockchainTx(tx: RegisterBlockchainTx): codecImpl.app.ITx {
+  return {
+    issueBlockchainNftMsg: codecImpl.blockchain.IssueTokenMsg.create({
+      id: Encoding.toUtf8(tx.blockchainId),
+      owner: decodeBnsAddress(keyToAddress(tx.signer)).data,
+      approvals: undefined,
+      details: codecImpl.blockchain.TokenDetails.create({
+        chain: codecImpl.blockchain.Chain.create({}),
```

Comment:
Why is this empty?

Reply:
Because we can ;) This is the smallest possible implementation accepted by the server

Reply:
After [splitting BnsBlockchainId and ChainId](https://github.com/iov-one/iov-core/pull/535/commits/f1aa56291987bd3ce62c09df146ba56f99a60a42) I now have a better idea what goes into `chain`. Will add some initial `blockchain.Chain` data now.

Reply:
Added in https://github.com/iov-one/iov-core/pull/535/commits/f265f1d2f363dab10c43f01dee377429d75cff5f